src/SonarScanner.MSBuild.Common/EnvironmentBasedPlatformHelper.cs

```diff
@@ -19,19 +19,25 @@
  */
 
 using System;
+using System.IO;
 
 namespace SonarScanner.MSBuild.Common;
 
 public class EnvironmentBasedPlatformHelper : IPlatformHelper
 {
+    private bool? isMacOs;
     public static IPlatformHelper Instance { get; } = new EnvironmentBasedPlatformHelper();
 
     private EnvironmentBasedPlatformHelper()
     {
     }
 
     public string GetFolderPath(Environment.SpecialFolder folder, Environment.SpecialFolderOption option) => Environment.GetFolderPath(folder, option);
-    public bool DirectoryExists(string path) => System.IO.Directory.Exists(path);
+    public bool DirectoryExists(string path) => Directory.Exists(path);
     public bool IsWindows() => Environment.OSVersion.Platform == PlatformID.Win32NT;
-    public bool IsMacOSX() => Environment.OSVersion.Platform == PlatformID.MacOSX;
+
+    // There's a more elegant way to obtain which operating system the app is running on. Unfortunately it's not suported in .NET Framework 4.6.2, only from 4.7.1
+    // SystemVersion.plist exists on Mac Os (and iOS) for a very long time (at least from 2002), so it's safe to check it, even though it's not a pretty solution.
+    // TODO: once we drop support for .NET Framework 4.6.2 replace this call with System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(OSPlatform.OSX)
+    public bool IsMacOs() => isMacOs ??= File.Exists(@"/System/Library/CoreServices/SystemVersion.plist");
```

Comment:
We multi-target:
https://github.com/SonarSource/sonar-scanner-msbuild/blob/8b80bc5b79cbe328aca51f2a4f349ed26f684d19/src/SonarScanner.MSBuild.Common/SonarScanner.MSBuild.Common.csproj#L3C54-L3C73

Does the `IsOSPlatform` check work with conditional compilation `#if netstandard1_1`?